web/src/api/images/delete_all_images.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  errorHasuraQuery,
+  errorNotAllowed,
+  errorResponse,
+} from "src/backend/errors";
+import { NextApiRequest, NextApiResponse } from "next";
+import { protectInternalEndpoint } from "src/backend/utils";
+import { getAPIServiceGraphqlClient } from "src/backend/graphql";
+import { getSdk as getUnverifiedImagesSDK } from "@/api/images/graphql/getUnverifiedImages.generated";
+import {
+  S3Client,
+  ListObjectsCommand,
+  PutObjectTaggingCommand,
+} from "@aws-sdk/client-s3";
+import { logger } from "@/lib/logger";
+
+export type DeleteAllImagesResponse = {
+  success?: boolean;
+};
+
+/**
+ * TODO: Triggered when user deletes an app. Sets all unverified images to expire in 3 days.
+ * @param req
+ * @param res
+ */
+export const handleDeleteAllImages = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  try {
+    if (!protectInternalEndpoint(req, res)) {
+      return;
+    }
+
+    if (req.method !== "POST") {
+      return errorNotAllowed(req.method!, res, req);
+    }
+
+    if (req.body.action?.name !== "delete_unverified_images") {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "Invalid action.",
+        code: "invalid_action",
+      });
+    }
+
+    const userId = req.body.session_variables["x-hasura-user-id"];
+    if (!userId) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "userId must be set.",
+        code: "required",
+      });
+    }
+
+    const teamId = req.body.session_variables["x-hasura-team-id"];
+    if (!teamId) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "teamId must be set.",
+        code: "required",
+      });
+    }
+    const app_id = req.body.input.app_id;
+    if (!app_id) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "app_id must be set.",
+        code: "required",
+      });
+    }
+    const client = await getAPIServiceGraphqlClient();
+    const { app: appInfo } = await getUnverifiedImagesSDK(
+      client
+    ).GetUnverifiedImages({
+      team_id: teamId,
+      app_id: app_id as string,
+      user_id: userId,
+    });
+    const userMembership = appInfo[0].team.memberships.find(
+      (membership) => membership.user_id === userId
+    );
+    // There should only be one app with the matching app_id and team_id
+    // Only Owner is allowed to delete
+    if (
+      appInfo.length === 0 ||
+      appInfo[0].app_metadata.length === 0 ||
+      !userMembership ||
+      userMembership.role !== "OWNER"
```

Comment:
another minor improvement, you could already account for roles in the query

Reply:
This query is used by getunverified images too. I added a roles check to the other query though